ktor-client/ktor-client-apache/jvm/src/io/ktor/client/engine/apache/ApacheEngine.kt

```diff
@@ -33,7 +33,13 @@ internal class ApacheEngine(override val config: ApacheEngineConfig) : HttpClien
     private fun prepareClient(): CloseableHttpAsyncClient {
         val clientBuilder = HttpAsyncClients.custom()
         with(clientBuilder) {
-            setThreadFactory { Thread(it, "Ktor-client-apache").apply { isDaemon = true } }
+            setThreadFactory {
+                Thread(it, "Ktor-client-apache").apply {
+                    isDaemon = true
+                    setUncaughtExceptionHandler { _, _ -> }
```

Comment:
Shouldn't we do something instead of ignoring exceptions?

Reply:
We report exceptions with our client coroutines and channels. Here we prevent to log cancellation exceptions unhandled by Apache.